quarkus/tests/integration/src/main/java/org/keycloak/it/utils/DockerKeycloakDistribution.java

```diff
@@ -0,0 +1,144 @@
+package org.keycloak.it.utils;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.jboss.logging.Logger;
+import org.keycloak.common.Version;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.ToStringConsumer;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.io.File;
+import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+
+public final class DockerKeycloakDistribution implements KeycloakDistribution {
+
+    private static final Logger LOGGER = Logger.getLogger(DockerKeycloakDistribution.class);
+
+    private boolean debug;
+    private boolean manualStop;
+    private int exitCode = -1;
+
+    private List<String> stdout = List.of();
+    private List<String> stderr = List.of();
+    private ToStringConsumer backupConsumer = new ToStringConsumer();
+
+    private File distributionFile = new File("../../../distribution/server-x-dist/target/keycloak.x-" + Version.VERSION_KEYCLOAK + ".tar.gz");
+
+    private GenericContainer keycloakContainer = null;
+
+    private GenericContainer runKeycloakContainer() {
+        if (!distributionFile.exists()) {
+            throw new RuntimeException("Distribution archive " + distributionFile.getAbsolutePath() +" doesn't exists");
+        }
+        // TODO: decide on one approach
```

Comment:
Here I'm providing the two implementations:
 - using a local `Dockerfile`
 - downloading the server-x `Dockerfile` from GH
 
@stianst can we consider to move the server-x `Dockerfile` to only this repository in the future?

I'm happy to keep only the preferred and agreed implementation BTW.